kernel_hardening_checker/config_files/distros/get_kconfigs.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Links with raw content
+links=(
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Android 14 (6.1) aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Debian 13 Trixie aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Debian 13 Trixie x86_64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/CentOS Hyperscale 9 aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/CentOS Hyperscale 9 x86_64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Fedora 39 Core aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Fedora 39 Core x86_64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Oracle Linux 9 (UEK-NEXT) aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Oracle Linux 9 (UEK-NEXT) x86_64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Ubuntu 24.04 LTS Noble aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Ubuntu 24.04 LTS Noble x86_64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Upstream Default 6.6 aarch64/config"
+    "https://raw.githubusercontent.com/oracle/kconfigs/refs/heads/gh-pages/out/Upstream Default 6.6 x86_64/config"
+)
+
+# Parse 'distro' from link and create 'distro'.config file in kernel_hardening_checker/config_files/distros
+for url in "${links[@]}"; do
+    folder_link="${url%/*}" # Strip 'config'
+    distro_name="${folder_link##*/}" # Extract distro name
+    cleaning_stage_2=$(echo ${distro_name// /_}) # Replace spaces with underscores
+    filename="${cleaning_stage_2}.config" # Create file name
+    target="${filename}"
+    wget -O "$target" "$url" # Fetch kconfig
+    mv "config" "${target}" #
+done
+
+# Fetch some other kconfigs
+wget -O Clearlinux_x86-64.config https://raw.githubusercontent.com/clearlinux-pkgs/linux/master/config
+wget -O Clearlinux_x86-64.cmdline https://raw.githubusercontent.com/clearlinux-pkgs/linux/master/cmdline
+wget -O OpenSUSE_x86-64.config https://raw.githubusercontent.com/openSUSE/kernel-source/refs/heads/master/config/x86_64/default
```

Comment:
It's an interesting question. The simple answer - the latest version. Previously there was a link to openSUSE-15.1: `https://kernel.opensuse.org/cgit/kernel-source/plain/config/x86_64/default?h=openSUSE-15.1`, but now I couldn't find a link to the kconfig of any specific version of openSUSE. It appears that the source code archives have been removed or relocated to a less accessible location.

On GitHub, there are two main branches: stable and master. The master branch is used in openSUSE Tumbleweed, which doesn't have any official releases and is updated constantly. The stable branch is used in the current openSUSE Leap version (it looks like 15.6 is the latest one).

Here is the source: https://kernel.suse.com/master.html

And the part of the actual kconfig related to the distro:

```
CONFIG_SUSE_KERNEL=y

#
# SUSE Release Details
#
# CONFIG_SUSE_PRODUCT_SLE is not set
# CONFIG_SUSE_PRODUCT_OPENSUSE_LEAP is not set
CONFIG_SUSE_PRODUCT_OPENSUSE_TUMBLEWEED=y
# CONFIG_SUSE_PRODUCT_ALP is not set
CONFIG_SUSE_PRODUCT_CODE=3
CONFIG_SUSE_VERSION=0
CONFIG_SUSE_PATCHLEVEL=0
CONFIG_SUSE_AUXRELEASE=0
# CONFIG_SUSE_KERNEL_SUPPORTED is not set
# CONFIG_SUSE_KERNEL_RELEASED is not set
# end of SUSE Release Details
```

By referring to the master, I couldn't really say what release is here, as it can be updated at any time.